test/int/puppeteer/puppeteerSupport.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/*
+* Functions that make puppeteer testing easier
+*/
+
+import * as util from 'util';
+import * as request from 'request';
+import * as puppeteer from 'puppeteer';
+
+/**
+ * Connect puppeteer to a currently running instance of chrome
+ * @param port The port on which the chrome debugger is running
+ */
+export async function connectPuppeteer(port: number): Promise<puppeteer.Browser> {
+
+    const resp = await util.promisify(request)(`http://localhost:${port}/json/version`);
+    const { webSocketDebuggerUrl } = JSON.parse(resp.body);
+
+    const browser = await puppeteer.connect({ browserWSEndpoint: webSocketDebuggerUrl, defaultViewport: null });
+    return browser;
+}
+
+/**
+ * Get the first (or only) page loaded in chrome
+ * @param browser Puppeteer browser object
+ */
+export async function firstPage(browser: puppeteer.Browser): Promise<puppeteer.Page> {
+    return browser.pages().then(p => p[0]);
```

Comment:
Something like this?

```suggestion
    return (await browser.pages())[0];
```

I think I originally had something like that, but thought using `then` was a little more explicit about what's happening (i.e. awaiting the result of `pages()` then returning the first item) .

>If we expect the browser to have a single page, it could make sense to verify this and throw an exception if we find more than one page. That could make it easier to diagnose what is happening if for some reason the browser ever has more than the single page we expect.

Thinking about it a bit more, a better approach is probably just to find the page we want via url, and then throw if we can't find it. It shouldn't matter if there are multiple pages if we can still run our tests correctly.

Reply:
That looks good

Reply:
Done